core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2020,6 +2020,14 @@ package object config {
     .toSequence
     .createWithDefault(Nil)
 
+  private[spark] val AUTO_STOP_ACTIVE_SPARK_CONTEXTS =
+    ConfigBuilder("spark.kubernetes.submit.autoStopActiveSparkContexts")
```

Comment:
This looks exceptional. Although I understand the situation, it's really weird to have `spark.kubernetes.` configuration in this file.

Reply:
I'd recommend to move this to K8s module and use config string name instead in `SparkSubmit.scala`.

Reply:
Do you mean having `kubernetes` within this file in the core module?

It follows the pattern of:
- spark.yarn.stagingDir
- spark.yarn.isPython
- spark.yarn.kerberos.renewal.excludeHadoopFileSystems
- spark.yarn.dist.pyFiles

Reply:
And there is:
- spark.standalone.submit.waitAppCompletion